### PR TITLE
OAK-10547 - Fix: Indexing job fails at the end of reindexing if it took more than 24h to run

### PR DESCRIPTION
**Pull request by nfsantos**
Fix support formatting durations longer than 24h.

### CODE REVIEW

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java [hunk 1]
```diff
@@ -18,18 +18,28 @@
...
-import java.time.LocalTime;
-import java.time.format.DateTimeFormatter;
 import java.util.concurrent.TimeUnit;
 
 public class FormattingUtils {
     public static String formatToSeconds(Stopwatch stopwatch) {
-        LocalTime seconds = LocalTime.ofSecondOfDay(stopwatch.elapsed(TimeUnit.SECONDS));
-        return DateTimeFormatter.ISO_TIME.format(seconds);
+        long seconds = stopwatch.elapsed(TimeUnit.SECONDS);
+        long absSeconds = Math.abs(seconds);
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java:26]: That way we don't detect if the time went backwards, which may be the case in some edge case.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java [hunk 2]
```diff
@@ -18,18 +18,28 @@
...
+        return String.format(
+                "%02d:%02d:%02d",
+                absSeconds / 3600,
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java]: The formulas look correct, but it should be possible to use Java utilities...

```
long millis = -123456789;
long hours = Math.abs(TimeUnit.MILLISECONDS.toHours(millis));
long minutes = Math.abs(TimeUnit.MILLISECONDS.toMinutes(millis)) % 60;
long seconds = Math.abs(TimeUnit.MILLISECONDS.toSeconds(millis)) % 60;
long milliseconds = Math.abs(millis) % 1000;

String formattedTime = String.format("%s%02d:%02d:%02d.%03d", millis < 0 ? "-" : "", hours, minutes, seconds, milliseconds);
System.out.println(formattedTime);
```

**Review by thomasmueller — CHANGES_REQUESTED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java (hunk 1 above)
Comment (nfsantos) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java:26]: Changed to print negative time spans: prefix with `-`.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java (hunk 2 above)
Comment (nfsantos) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java]: Yes, it becomes easier to read. I have refactored the code.

File: oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ */
+package org.apache.jackrabbit.oak.plugins.index;
+
+import org.apache.jackrabbit.guava.common.base.Stopwatch;
+import org.apache.jackrabbit.guava.common.base.Ticker;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.*;
```
Comment (fabriziofortino) [inline oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtilsTest.java]: avoid `*` imports

**Review by fabriziofortino — APPROVED** [review]

**Review by thomasmueller — APPROVED** [review]

**Pull request merged**